src/components/StatsBoxGrid/useStatsBoxGrid.tsx

```diff
@@ -1,259 +1,100 @@
+    maximumSignificantDigits: 4,
+  }).format(txs)
 }
 
-interface IFetchTxResponse {
-  unixTimeStamp: string
-  transactionCount: number
+const formatNodes = (nodes: number, locale: string): string => {
+  return new Intl.NumberFormat(locale, {
+    minimumSignificantDigits: 3,
+    maximumSignificantDigits: 4,
+  }).format(nodes)
 }
 
-export const useStatsBoxGrid = () => {
+export const useStatsBoxGrid = ({
+  totalEthStaked,
+  nodeCount,
+  totalValueLocked,
+  txCount,
+}: AllMetricData): StatsBoxMetric[] => {
   const { t } = useTranslation("page-index")
   const { locale } = useRouter()
 
-  const [totalEthStaked, setTotalEthStaked] = useState<State>({
-    data: [],
-    value: "0",
-    hasError: false,
-  })
-  const [valueLocked, setValueLocked] = useState<State>({
-    data: [],
-    value: "0",
-    hasError: false,
-  })
-  const [txs, setTxs] = useState<State>({
-    data: [],
-    value: "0",
-    hasError: false,
-  })
-  const [nodes, setNodes] = useState<State>({
-    data: [],
-    value: "0",
-    hasError: false,
-  })
   const [selectedRangeTotalStaked, setSelectedRangeTotalStaked] =
-    useState<string>(ranges[0])
-  const [selectedRangeTvl, setSelectedRangeTvl] = useState<string>(ranges[0])
+    useState<string>(RANGES[0])
+  const [selectedRangeTvl, setSelectedRangeTvl] = useState<string>(RANGES[0])
   const [selectedRangeNodes, setSelectedRangeNodes] = useState<string>(
-    ranges[0]
+    RANGES[0]
   )
-  const [selectedRangeTxs, setSelectedRangeTxs] = useState<string>(ranges[0])
-
-  useEffect(() => {
-    const localeForStatsBoxNumbers = getLocaleForNumberFormat(locale! as Lang)
-
-    const formatTotalStaked = (amount: number): string => {
-      return new Intl.NumberFormat(localeForStatsBoxNumbers, {
-        notation: "compact",
-        minimumSignificantDigits: 3,
-        maximumSignificantDigits: 4,
-      }).format(amount)
-    }
+  const [selectedRangeTxs, setSelectedRangeTxs] = useState<string>(RANGES[0])
 
-    const formatTVL = (tvl: number): string => {
-      return new Intl.NumberFormat(localeForStatsBoxNumbers, {
-        style: "currency",
-        currency: "USD",
-        notation: "compact",
-        minimumSignificantDigits: 3,
-        maximumSignificantDigits: 4,
-      }).format(tvl)
-    }
+  const localeForNumberFormat = getLocaleForNumberFormat(locale! as Lang)
 
-    const formatTxs = (txs: number): string => {
-      return new Intl.NumberFormat(localeForStatsBoxNumbers, {
-        notation: "compact",
-        minimumSignificantDigits: 3,
-        maximumSignificantDigits: 4,
-      }).format(txs)
-    }
-
-    const formatNodes = (nodes: number): string => {
-      return new Intl.NumberFormat(localeForStatsBoxNumbers, {
-        minimumSignificantDigits: 3,
-        maximumSignificantDigits: 4,
-      }).format(nodes)
-    }
-
-    const weiToRoundedEther = (wei: number): number => Math.floor(wei * 1e-18)
-
-    const fetchTotalStaked = async (): Promise<void> => {
```

Comment:
The four data fetching functions were move to `lib/api/*` to be accessed by the _page_ component